Implement proper support for nested complex env values (#4216)
pydantic/pydantic: changes/4216-raistleitner.md

> Pydantic didn't support setting fields containing "complex values" like dicts or lists from environment variables using nesting. This PR contains an implementation addressing this issue by checking which field the nested environment variable targets to, checking if the field is a complex value and if so, trying to parse the environment value as a JSON.

Reviewer: I would suggest changing it to a simple sentence.
```suggestion
Add support for setting nested fields containing complex values like dict or list from environment variables
```
